lib/rage/controller/api.rb

```diff
@@ -331,4 +331,26 @@ def params
       @params ||= ActionController::Parameters.new(@__params)
     end
   end
+
+  # Checks if the request is stale to decide if the action has to be rendered or the cached version is still valid. Use this method to implement conditional GET.
+  #
+  # @param etag [String] The etag of the requested resource.
+  # @param last_modified [Time] The last modified time of the requested resource.
```

Comment:
Why do you think `String` is more reasonable here?
If we want to do what Rails does, I think we should [stick to time](https://apidock.com/rails/ActionController/ConditionalGet/fresh_when).

On another note, this brought to my attention that I definitely need to convert to `Time` [here](https://github.com/rage-rb/rage/pull/55/files#diff-3e7ad6582a85dbe56681557a219a4c3f96181cf138ad2b169d9455b30160c9d3R27) and that the tests should probably pass strings as HTTP headers to reflect the real world.

Reply:
Right, I think `Time` makes more sense here, but [this line](https://github.com/rage-rb/rage/pull/55/files#diff-3e7ad6582a85dbe56681557a219a4c3f96181cf138ad2b169d9455b30160c9d3R56) will fail if you use `Time` with the current implementation.

Reply:
> Right, I think Time makes more sense here, but [this line](https://github.com/rage-rb/rage/pull/55/files#diff-3e7ad6582a85dbe56681557a219a4c3f96181cf138ad2b169d9455b30160c9d3R56) will fail if you use Time with the current implementation.

Not sure what you're referring to here.
[b9c74ad](https://github.com/rage-rb/rage/pull/55/commits/b9c74ad161160d1034e124b1bbc1337bae3cfe07) now takes into account the fact that HTTP headers are always strings and converts it to `Time`. If the user is not passing a wrong type to `stale?`, the line you mentioned should work.
We could think about adding a type conversion to `stale?`, something like:
```ruby
def stale?(etag:, last_modified:)
  last_modified = Time.new(last_modified.to_s)
  # ...
end
```

This would allow the user to pass in both `String` and `Time` objects, which would add more flexibility but also bloat the code with conversion logic.
What do you think?

Reply:
I think allowing just `Time` is sufficient.